src/protocols/user_directed_commissioning/UserDirectedCommissioningServer.cpp

```diff
@@ -242,25 +242,56 @@ CHIP_ERROR IdentificationDeclaration::ReadPayload(uint8_t * udcPayload, size_t p
             mRotatingIdLen = reader.GetLength();
             err            = reader.GetBytes(mRotatingId, sizeof(mRotatingId));
             break;
-        case kAppVendorIdListTag:
+        case kTargetAppListTag:
             // app vendor list
             {
+                ChipLogError(AppServer, "TLV found an applist");
                 chip::TLV::TLVType listContainerType = chip::TLV::kTLVType_List;
                 ReturnErrorOnFailure(reader.EnterContainer(listContainerType));
 
                 while ((err = reader.Next()) == CHIP_NO_ERROR && mNumAppVendorIds < sizeof(mAppVendorIds))
                 {
                     containerTag = reader.GetTag();
                     tagNum       = static_cast<uint8_t>(chip::TLV::TagNumFromTag(containerTag));
```

Comment:
Pre-existing, but TagNumFromTag can return garbage if the tag type has not been checked.

Reply:
added checks